plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdMetaModel.java

```diff
@@ -128,11 +128,77 @@ public boolean supportsTriggers(@NotNull GenericDataSource dataSource) {
         return true;
     }
 
+    @Override
+    public JDBCStatement prepareTableTriggersLoadStatement(@NotNull JDBCSession session, @NotNull GenericStructContainer container, @Nullable GenericTableBase table) throws SQLException {
+        JDBCPreparedStatement dbStat = session.prepareStatement(
+                "SELECT RDB$TRIGGER_NAME AS TRIGGER_NAME, RDB$RELATION_NAME AS OWNER, T.* FROM RDB$TRIGGERS T\n" +
+                        "WHERE RDB$RELATION_NAME" + (table == null ? " IS NOT NULL" : "=?"));
+        if (table != null) {
+            dbStat.setString(1, table.getName());
+        }
+        return dbStat;
+    }
+
+    @Override
+    public GenericTrigger createTableTriggerImpl(@NotNull JDBCSession session, @NotNull GenericStructContainer container, @NotNull GenericTableBase parent, String triggerName, @NotNull JDBCResultSet dbResult) throws DBException {
+        if (CommonUtils.isEmpty(triggerName)) {
+            triggerName = JDBCUtils.safeGetStringTrimmed(dbResult, "RDB$TRIGGER_NAME");
+        }
+        if (triggerName == null) {
+            return null;
+        }
+        int sequence = JDBCUtils.safeGetInt(dbResult, "RDB$TRIGGER_SEQUENCE");
+        int type = JDBCUtils.safeGetInt(dbResult, "RDB$TRIGGER_TYPE");
+        String description = JDBCUtils.safeGetStringTrimmed(dbResult, "RDB$DESCRIPTION");
+        int systemFlag = JDBCUtils.safeGetInt(dbResult, "RDB$SYSTEM_FLAG");
+        boolean isSystem = true;
+        if (systemFlag == 0) { // System flag value 0 - if user-defined and 1 or more if system
+            isSystem = false;
+        }
```

Comment:
Can be inlined because it looks prettier to me that way (IMHO):

```java
// System flag value 0 - if user-defined and 1 or more if system
boolean isSystem = systemFlag > 0;
```

Instead, you can do this directly in the QUERY itself:
```sql
SELECT ... RDB$SYSTEM_FLAG > 0 AS IS_SYSTEM ...
```